lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,181 @@
+      run_playbook_command(deployment_master, user, "ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/rhel_subscribe.yml -i "\
+                         "/usr/share/ansible/openshift-ansible/rhel_subscribe_inventory.yaml 1> /tmp/ansible.log 2> /tmp/openshift-ansible.log.2 < /dev/null")
+    end
+
+    def subscription_needed?
+      container_deployment_nodes.count > 1
+    end
+
+    def rhsm_creds
+      [rhsm_auth.userid, rhsm_auth.password, rhsm_auth.rhsm_sku]
+    end
+
+    def ssh_user
+      ssh_auth.userid
+    end
+
+    def add_deployment_provider(options)
+      object_send(:add_deployment_provider, options)
+    end
+
+    def regenerate_ansible_inventory
+      object_send(:generate_ansible_yaml)
+    end
+
+    def regenerate_ansible_subscription_inventory
+      object_send(:generate_ansible_inventory_for_subscription)
+    end
+
+    def add_automation_task(task)
+      ar_method do
+        @object.automation_task = AutomationTask.find_by_id(task.id)
+        @object.save!
+      end
+    end
+
+    def customize(options = {})
+      ar_method do
+        if @object.customizations.empty?
+          @object.customizations = {:agent => {}}
+        end
+        options.each do |key, val|
+          @object.customizations[:agent][key] = val
+        end
+        @object.save!
+      end
+    end
+
+    def perform_scp(ip, username, local_path, remote_path)
+      Net::SCP.upload!(ip,
+                       username,
+                       local_path,
+                       remote_path,
+                       :ssh => {:key_data => ssh_auth.auth_key})
+    end
+
+    def perform_agent_commands(ip, username, commands = [])
+      ssh = LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      ssh.perform_commands(commands)
+    end
+
+    def check_connection(deployment_host_ip, username, nodes_ips)
+      agent = LinuxAdmin::SSHAgent.new(ssh_auth.auth_key, "/tmp/ssh_manageiq/ssh_manageiq_#{id}")
+      success, unreachable_ips = agent.with_service do |socket|
+        ssh = LinuxAdmin::SSH.new(deployment_host_ip, username)
+        unreachable_ips = nodes_ips.select do |sub_ip|
+          !ssh.perform_commands(["sudo -E ssh -o 'StrictHostKeyChecking no' #{username}@#{sub_ip} echo $?"], socket)[:stdout].include?("0\r\n")
+        end
+        [unreachable_ips.empty?, unreachable_ips]
+      end
+      unless success
+        raise StandardError, "couldn't connect to : #{unreachable_ips.join(',')}"
+      end
+      success
+    end
+
+    def playbook_running?
+      pid = ar_method do
+        @object.customizations[:agent][:ansible_pid] if @object.customizations[:agent]
+      end
+      !pid.nil?
+    end
+
+    def run_playbook_command(ip, username, cmd)
+      result = {:finished => false}
+      ansible_pid = ar_method do
+        @object.customizations[:agent][:ansible_pid] if @object.customizations[:agent]
```

Comment:
`ar_method` everywhere is noise.
Consider a helper method to read `@object.customizations`.